new resource "azurerm_spring_cloud_java_deployment" and "azurerm_spring_cloud_active_deployment" (#9959)
terraform-providers/terraform-provider-azurerm: website/docs/r/spring_cloud_java_deployment.html.markdown

---
subcategory: "Spring Cloud"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_spring_cloud_java_deployment"
description: |-

>   Manage an Azure Spring Cloud Deployment which runtime is Java.
> ---
> 
> # azurerm_spring_cloud_java_deployment
> 
> Manage an Azure Spring Cloud Deployment which runtime is Java.

Reviewer: ```suggestion
  Manages an Azure Spring Cloud Deployment which runtime is Java.
---

# azurerm_spring_cloud_java_deployment

Manages an Azure Spring Cloud Deployment which runtime is Java.
```

## Example Usage

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_spring_cloud_service" "example" {
  name                = "example-springcloud"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
}

resource "azurerm_spring_cloud_app" "example" {
  name                = "example-springcloudapp"
  resource_group_name = azurerm_resource_group.example.name
  service_name        = azurerm_spring_cloud_service.example.name

  identity {
    type = "SystemAssigned"
  }
}

resource "azurerm_spring_cloud_java_deployment" "example" {
  name                = "deploy1"
  spring_cloud_app_id = azurerm_spring_cloud_app.example.id
  cpu                 = 2
  memory_in_gb        = 4
  instance_count      = 2
  jvm_options         = "-XX:+PrintGC"

>   runtime_version     = "Java_8"

Reviewer: As above
```suggestion
  runtime_version     = "Java_11"
```

>   env = {
>     "Env" : "Staging"
>   }

Reviewer: Perhaps some additional entries here to help users?
```suggestion
  env = {
    "Env"    = "Staging"
    "my_var" = "someValue"
  }
```

}
```

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the Spring Cloud Deployment. Changing this forces a new resource to be created.

* `spring_cloud_app_id` - (Required) Specifies the id of the Spring Cloud Application in which to create the Deployment. Changing this forces a new resource to be created.

> * `cpu` - (Optional) Specifies the required cpu of the Spring Cloud Deployment. Possible Values are between `1` and `4`.

Reviewer: ```suggestion
* `cpu` - (Optional) Specifies the required cpu of the Spring Cloud Deployment. Possible Values are between `1` and `4`. Defaults to `1` if not specified.
```

> * `memory_in_gb` - (Optional) Specifies the required memory size of the Spring Cloud Deployment. Possible Values are between `1` and `8`.

Reviewer: ```suggestion
* `memory_in_gb` - (Optional) Specifies the required memory size of the Spring Cloud Deployment. Possible Values are between `1` and `8`. Defaults to `1` if not specified. 
```

> * `instance_count` - (Optional) Specifies the required instance count of the Spring Cloud Deployment. Possible Values are between `1` and `500`.

Reviewer: ```suggestion
* `instance_count` - (Optional) Specifies the required instance count of the Spring Cloud Deployment. Possible Values are between `1` and `500`. Defaults to `1` if not specified. 
```

* `jvm_options` - (Optional) Specifies the jvm option of the Spring Cloud Deployment.

* `runtime_version` - (Optional) Specifies the runtime version of the Spring Cloud Deployment. Possible Values are `Java_8` and `Java_11`. Defaults to `Java_8`.

> * `env` - (Optional) Specifies the environment variables of the Spring Cloud Deployment.

Reviewer: ```suggestion
* `env` - (Optional) Specifies the environment variables of the Spring Cloud Deployment as a map of key-value pairs.
```

---
subcategory: "Spring Cloud"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_spring_cloud_java_deployment"
description: |-

>   Manages an Azure Spring Cloud Deployment which runtime is Java.

Reviewer: ```suggestion
  Manages an Azure Spring Cloud Deployment with a Java runtime.
```

---

# azurerm_spring_cloud_java_deployment

> Manages an Azure Spring Cloud Deployment which runtime is Java.

Reviewer: ```suggestion
Manages an Azure Spring Cloud Deployment with a Java runtime.
```
